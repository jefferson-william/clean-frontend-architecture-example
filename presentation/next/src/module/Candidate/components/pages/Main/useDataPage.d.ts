import { DataProps } from '~/types/data'

export interface CandidateSelectorProps {
  formData: DataProps
  obtainedUserDataFromLinkedin: boolean
}
