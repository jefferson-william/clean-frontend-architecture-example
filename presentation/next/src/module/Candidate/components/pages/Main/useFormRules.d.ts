import { DataProps } from '~/types/data'

export interface UseFormRulesProps {
  lastPanel: boolean
  formData: DataProps
}
