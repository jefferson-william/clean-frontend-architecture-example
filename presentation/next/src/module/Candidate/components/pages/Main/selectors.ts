import { screen } from '~/shared/utils/test-utils'

export function findByKnowledgeInput() {
  return screen.findByRole('textbox', { name: 'Conhecimentos' })
}

export function getAllKnowledgeInput() {
  return screen.getAllByRole('textbox', { name: 'Conhecimentos' })
}
