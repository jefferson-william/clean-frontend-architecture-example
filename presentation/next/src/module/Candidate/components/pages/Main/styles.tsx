import styled from 'styled-components'
import { Container } from '@material-ui/core'

export const Main = styled(Container)`
  width: auto;
  margin-top: 48px;
  .main {
    &__tabs {
      margin-bottom: 48px;
    }
    &__tab-panel {
      margin: 2px;
    }
    &__card {
      display: flex;
      flex-direction: column;
      padding: 48px;
    }
    &__form-control:not(:first-child) {
      margin-top: 32px;
    }
    &__actions {
      display: flex;
      justify-content: center;
      margin-top: 48px;
    }
    &__button:not(:first-child) {
      margin-left: 16px;
    }
    &__linkedin-area {
      display: flex;
      flex-direction: column;
      align-items: center;
      height: 73px;
      margin-bottom: 48px;
    }
    &__linkedin-text {
      margin-bottom: 16px;
    }
    &__linkedin-image {
      max-width: 180px;
    }
    &__linkedin-loader {
      width: 180px;
      height: 33px;
    }
    &__avatar {
      margin: 0 auto;
    }
  }
`
