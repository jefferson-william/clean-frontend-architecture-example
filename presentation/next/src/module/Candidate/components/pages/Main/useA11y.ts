import { useCallback } from 'react'

const useA11y = () => {
  const a11yProps = useCallback(
    (index: number) => ({
      id: `full-width-tab-${index}`,
      'aria-controls': `full-width-tabpanel-${index}`,
    }),
    []
  )

  return {
    a11yProps,
  }
}

export default useA11y
