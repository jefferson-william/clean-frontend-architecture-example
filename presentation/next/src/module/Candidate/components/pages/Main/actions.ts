import { getAllKnowledgeInput } from '~/next/module/Candidate/components/pages/Main/selectors'
import { MyRenderResult, screen, userEvent } from '~/next/shared/utils/test-utils'

export function clickNextButton() {
  userEvent.click(screen.getByRole('button', { name: 'Pr√≥ximo' }))
}

export function clickSendButton() {
  userEvent.click(screen.getByRole('button', { name: 'Enviar' }))
}

export function clickWhereDidWorkAddButton() {
  userEvent.click(screen.getByRole('button', { name: 'Adicionar' }))
}

export function populateBasicDataForm() {
  userEvent.type(screen.getByRole('textbox', { name: 'Nome completo' }), 'Steve Jobs')
  userEvent.type(screen.getByRole('textbox', { name: 'E-mail' }), 'steve.jobs@email.com')
}

export async function populateKnowledgeForm(wrapper: MyRenderResult) {
  userEvent.type(getAllKnowledgeInput()[0], 'NodeJS')

  clickWhereDidWorkAddButton()

  await wrapper.findByInputName('knowledge[1]')

  userEvent.type(getAllKnowledgeInput()[1], 'React')
}
