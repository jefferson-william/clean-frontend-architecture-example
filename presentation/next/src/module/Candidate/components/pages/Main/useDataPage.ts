import { useMemo, useState } from 'react'
import { useForm } from 'react-hook-form'
import { useSelector } from 'react-redux'
import { useTheme } from '@material-ui/core/styles'
import { CandidateSelectorProps } from '~/module/Candidate/components/pages/Main/useDataPage.d'
import States from '~/shared/store/rootStates.d'

const useDataPage = () => {
  const theme = useTheme()
  const { register, handleSubmit, control } = useForm()
  const { formData, obtainedUserDataFromLinkedin } = useSelector<States, CandidateSelectorProps>((state) => ({
    formData: state.Candidate.formData,
    obtainedUserDataFromLinkedin: state.Candidate.obtainedUserDataFromLinkedin,
  }))
  const [, setLinkedinAuthorizationToken] = useState<string>('')

  const firstPanel = useMemo(() => formData.panelIndex === 0, [formData.panelIndex])

  const lastPanel = useMemo(() => formData.panelIndex === 2, [formData.panelIndex])

  return {
    theme,
    firstPanel,
    lastPanel,
    formData,
    obtainedUserDataFromLinkedin,
    control,
    register,
    handleSubmit,
    setLinkedinAuthorizationToken,
  }
}

export default useDataPage
