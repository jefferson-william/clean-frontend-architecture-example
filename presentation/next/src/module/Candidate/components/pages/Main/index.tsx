import { FC } from 'react'
import { Experiment, Variant } from 'react-optimize'
import dynamic from 'next/dynamic'
import Image from 'next/image'
import { Avatar, Button, Card, Tab, Tabs, TextField, Typography } from '@material-ui/core'
import TabPanel from '~/next/module/Candidate/components/atoms/TabPanel'
import AddInformationFields from '~/next/module/Candidate/components/molecules/AddInformationFields'
import { Main } from '~/next/module/Candidate/components/pages/Main/styles'
import linkedinImage from '~/next/shared/assets/images/linkedin.png'
import Layout from '~/next/shared/components/templates/Layout'
import { LINKEDIN_IS_SHOW_VARIATION, OPTIMIZE_EXPERIMENTS } from '~/next/shared/constants'
import useA11y from './useA11y'
import useDataPage from './useDataPage'
import useFormRules from './useFormRules'
import useLinkedinRules from './useLinkedinRules'
import usePageRules from './usePageRules'

const LinkedIn = dynamic<any>(() => import('react-linkedin-login-oauth2'), { ssr: false })

const MainPage: FC = () => {
  const {
    theme,
    firstPanel,
    lastPanel,
    formData,
    obtainedUserDataFromLinkedin,
    control,
    register,
    handleSubmit,
    setLinkedinAuthorizationToken,
  } = useDataPage()
  const { handleLinkedinSuccess, handleLinkedinFailure } = useLinkedinRules({
    formData,
    control,
    obtainedUserDataFromLinkedin,
    setLinkedinAuthorizationToken,
  })
  const { handleChange, handleBack } = usePageRules({
    formData,
  })
  const { setWhereDidYouWork, setKnowledge, onSubmit } = useFormRules({ lastPanel, formData })
  const { a11yProps } = useA11y()

  return (
    <Layout>
      <Main maxWidth="sm">
        <Experiment id={OPTIMIZE_EXPERIMENTS.LINKEDIN_IS_SHOW}>
          <Variant id={LINKEDIN_IS_SHOW_VARIATION.Test}>
            <Tabs className="main__tabs" value={formData.panelIndex} onChange={handleChange} aria-label="Perguntas">
              <Tab label="Dados básicos" {...a11yProps(0)} />
              <Tab label="Onde já trabalhou" {...a11yProps(1)} />
              <Tab label="Conhecimentos" {...a11yProps(2)} />
            </Tabs>
          </Variant>
        </Experiment>
        <div className="main__linkedin-area">
          <Typography className="main__linkedin-text" variant="caption">
            Clique para preencher automaticamente
          </Typography>
          <LinkedIn
            clientId={process.env.NEXT_PUBLIC_LINKEDIN_CLIENT_ID}
            onSuccess={handleLinkedinSuccess}
            onFailure={handleLinkedinFailure}
            redirectUri={`${process.env.NEXT_PUBLIC_URL}/linkedin`}
            scope={process.env.NEXT_PUBLIC_LINKEDIN_CLIENT_SCOPE}
          >
            <Image
              src={linkedinImage}
              alt="Log in with Linked In"
              width="180"
              height="33"
              className="main__linkedin-image"
            />
          </LinkedIn>
        </div>
        <form onSubmit={handleSubmit(onSubmit)}>
          <TabPanel className="main__tab-panel" value={formData.panelIndex} index={0} dir={theme.direction}>
            <Card className="main__card">
              <Avatar
                className="main__avatar"
                alt="Foto do usuário"
                src={formData.picture}
                imgProps={{ width: 40, height: 40 }}
              />
              <TextField
                id="fullName"
                name="fullName"
                label="Nome completo"
                className="main__form-control"
                defaultValue={formData.fullName}
                inputRef={register({ required: true })}
                InputLabelProps={{ shrink: true }}
                required
              />
              <TextField
                id="email"
                name="email"
                type="email"
                label="E-mail"
                className="main__form-control"
                defaultValue={formData.email}
                inputRef={register({ required: true })}
                InputLabelProps={{ shrink: true }}
                required
              />
            </Card>
          </TabPanel>
          <TabPanel className="main__tab-panel" value={formData.panelIndex} index={1} dir={theme.direction}>
            <Card className="main__card">
              <AddInformationFields
                list={formData.whereDidYouWork}
                name="whereDidYouWork"
                text="Onde já trabalhou?"
                control={control}
                setList={setWhereDidYouWork}
                register={register}
                formControlClass="main__form-control"
              />
            </Card>
          </TabPanel>
          <TabPanel className="main__tab-panel" value={formData.panelIndex} index={2} dir={theme.direction}>
            <Card className="main__card">
              <AddInformationFields
                list={formData.knowledge}
                name="knowledge"
                text="Conhecimentos"
                control={control}
                setList={setKnowledge}
                register={register}
                formControlClass="main__form-control"
              />
            </Card>
          </TabPanel>
          <div className="main__actions">
            <Button className="main__button" variant="contained" disabled={firstPanel} onClick={handleBack}>
              Anterior
            </Button>
            <Button className="main__button" type="submit" variant="contained" color="primary">
              {formData.panelIndex === 2 ? 'Enviar' : 'Próximo'}
            </Button>
          </div>
        </form>
      </Main>
    </Layout>
  )
}

export default MainPage
