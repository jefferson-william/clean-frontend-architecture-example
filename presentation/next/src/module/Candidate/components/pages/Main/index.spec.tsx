import { populateWhereDidYouWorkForm } from '~/next/module/Candidate/components/molecules/AddInformationFields/actions'
import {
  findByWhereDidYouWorkInput,
  getAllWhereDidYouWorkInput,
} from '~/next/module/Candidate/components/molecules/AddInformationFields/selectors'
import Main from '~/next/module/Candidate/components/pages/Main'
import {
  clickNextButton,
  clickSendButton,
  populateBasicDataForm,
  populateKnowledgeForm,
} from '~/next/module/Candidate/components/pages/Main/actions'
import { findByKnowledgeInput } from '~/next/module/Candidate/components/pages/Main/selectors'
import { render, act, MyRenderResult, waitForElementToBeRemoved } from '~/next/shared/utils/test-utils'
import { MyRenderedOptions } from '~/next/shared/utils/test-utils/index.d'
import '~/next/__mocks__/nextRouter'

describe('pages/main', () => {
  let wrapper: MyRenderResult
  let options: MyRenderedOptions

  beforeEach(async () => {
    ;[wrapper, options] = await render(<Main />)
  })

  it('should to match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('should be able to perform the entire flow of filling out the form', async () => {
    await act(async () => {
      populateBasicDataForm()

      clickNextButton()

      await findByWhereDidYouWorkInput()

      await populateWhereDidYouWorkForm(wrapper)

      clickNextButton()

      await waitForElementToBeRemoved(() => getAllWhereDidYouWorkInput()[0])

      await findByKnowledgeInput()

      await populateKnowledgeForm(wrapper)

      clickSendButton()
    })

    expect(options.store.getState().Candidate.formData).toEqual({
      fullName: 'Steve Jobs',
      email: 'steve.jobs@email.com',
      whereDidYouWork: ['Amazon', 'Yahoo'],
      knowledge: ['NodeJS', 'React'],
      panelIndex: 2,
      picture: '',
    })
  })
})
