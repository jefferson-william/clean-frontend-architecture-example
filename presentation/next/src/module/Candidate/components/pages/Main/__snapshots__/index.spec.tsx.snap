// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/main should to match snapshot 1`] = `
<DocumentFragment>
  .c2 {
  width: auto;
  margin-top: 48px;
}

.c2 .main__tabs {
  margin-bottom: 48px;
}

.c2 .main__tab-panel {
  margin: 2px;
}

.c2 .main__card {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 48px;
}

.c2 .main__form-control:not(:first-child) {
  margin-top: 32px;
}

.c2 .main__actions {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 48px;
}

.c2 .main__button:not(:first-child) {
  margin-left: 16px;
}

.c2 .main__linkedin-area {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 73px;
  margin-bottom: 48px;
}

.c2 .main__linkedin-text {
  margin-bottom: 16px;
}

.c2 .main__linkedin-image {
  max-width: 180px;
}

.c2 .main__linkedin-loader {
  width: 180px;
  height: 33px;
}

.c2 .main__avatar {
  margin: 0 auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c0 .layout__header-loader {
  height: 66px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 66px;
}

.c1 .header__logo {
  max-width: 83px;
  height: 13px;
}

<div
    class="c0 layout"
  >
    <div
      class="MuiPaper-root c1 header MuiPaper-elevation4 MuiPaper-rounded"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iMTMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="Logo Intera"
            class="header__logo"
            decoding="async"
            src="/_next/image?url=%2Flogo-intera.png&w=256&q=75"
            srcset="/_next/image?url=%2Flogo-intera.png&w=96&q=75 1x, /_next/image?url=%2Flogo-intera.png&w=256&q=75 2x"
            style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiContainer-root c2 MuiContainer-maxWidthSm"
    >
      <div>
        <div>
          <div
            class="MuiTabs-root main__tabs"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                aria-label="Perguntas"
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="full-width-tabpanel-0"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
                  id="full-width-tab-0"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Dados básicos
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="full-width-tabpanel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                  id="full-width-tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Onde já trabalhou
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="full-width-tabpanel-2"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
                  id="full-width-tab-2"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    Conhecimentos
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="main__linkedin-area"
      >
        <span
          class="MuiTypography-root main__linkedin-text MuiTypography-caption"
        >
          Clique para preencher automaticamente
        </span>
        <button
          style="background: none; padding: 0px; cursor: pointer; outline: inherit;"
          type="button"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgwIiBoZWlnaHQ9IjMzIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Log in with Linked In"
              class="main__linkedin-image"
              decoding="async"
              src="/_next/image?url=%2Flinkedin.png&w=384&q=75"
              srcset="/_next/image?url=%2Flinkedin.png&w=256&q=75 1x, /_next/image?url=%2Flinkedin.png&w=384&q=75 2x"
              style="visibility: inherit; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </button>
      </div>
      <form>
        <div
          aria-labelledby="full-width-tab-0"
          class="main__tab-panel"
          dir="ltr"
          id="full-width-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="MuiPaper-root MuiCard-root main__card MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle main__avatar MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root main__form-control"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required"
                data-shrink="true"
                for="fullName"
                id="fullName-label"
              >
                Nome completo
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="fullName"
                  name="fullName"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root main__form-control"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required"
                data-shrink="true"
                for="email"
                id="email-label"
              >
                E-mail
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  id="email"
                  name="email"
                  required=""
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          aria-labelledby="full-width-tab-1"
          class="main__tab-panel"
          dir="ltr"
          hidden=""
          id="full-width-tabpanel-1"
          role="tabpanel"
        />
        <div
          aria-labelledby="full-width-tab-2"
          class="main__tab-panel"
          dir="ltr"
          hidden=""
          id="full-width-tabpanel-2"
          role="tabpanel"
        />
        <div
          class="main__actions"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained main__button Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Anterior
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained main__button MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Próximo
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
